src/workers/tx_sender.rs

```diff
@@ -42,53 +50,54 @@ impl Default for TxProps {
 impl TxSender {
     pub fn new(tpu_manager: Arc<TpuManager>) -> Self {
         Self {
-            enqueued_txs: Default::default(),
             tpu_manager,
             txs_sent: Default::default(),
         }
     }
-    /// en-queue transaction if it doesn't already exist
-    pub async fn enqnueue_tx(&self, sig: String, raw_tx: WireTransaction) {
-        self.enqueued_txs.write().await.push((sig, raw_tx));
-    }
 
     /// retry enqued_tx(s)
-    pub async fn forward_txs(&self, tx_batch_size: usize) {
-        if self.enqueued_txs.read().await.is_empty() {
+    async fn forward_txs(
+        &self,
+        sigs_and_slots: Vec<(String, u64)>,
+        txs: Vec<WireTransaction>,
+        postgres: Option<PostgresMpscSend>,
+    ) {
+        assert_eq!(sigs_and_slots.len(), txs.len());
+
+        if sigs_and_slots.is_empty() {
             return;
         }
 
-        let mut enqueued_txs = Vec::new();
-        std::mem::swap(&mut enqueued_txs, &mut *self.enqueued_txs.write().await);
-
-        let mut tx_remaining = enqueued_txs.len();
-        let mut enqueued_txs = enqueued_txs.into_iter();
         let tpu_client = self.tpu_manager.clone();
         let txs_sent = self.txs_sent.clone();
 
         tokio::spawn(async move {
-            while tx_remaining != 0 {
-                let mut batch = Vec::with_capacity(tx_batch_size);
-                let mut sigs = Vec::with_capacity(tx_batch_size);
-
-                for (batched, (sig, tx)) in enqueued_txs.by_ref().enumerate() {
-                    batch.push(tx);
-                    sigs.push(sig);
-                    tx_remaining -= 1;
-                    if batched == tx_batch_size {
-                        break;
+            let quic_response = match tpu_client.try_send_wire_transaction_batch(txs).await {
+                Ok(_) => {
+                    for (sig, _) in &sigs_and_slots {
+                        txs_sent.insert(sig.to_owned(), TxProps::default());
                     }
+                    1
                 }
-
-                match tpu_client.try_send_wire_transaction_batch(batch).await {
-                    Ok(_) => {
-                        for sig in sigs {
-                            txs_sent.insert(sig, TxProps::default());
-                        }
-                    }
-                    Err(err) => {
-                        warn!("{err}");
-                    }
+                Err(err) => {
+                    warn!("{err}");
+                    0
+                }
+            };
+
+            if let Some(postgres) = postgres {
+                for (sig, recent_slot) in sigs_and_slots {
+                    postgres
+                        .send(PostgresMsg::PostgresTx(PostgresTx {
+                            signature: sig.clone(),
```

Comment:
This should not be done here, but in block subscription stage where we iterate through all the transactions to find if they are in the block or not.